File: java2csharp/java/Samsung/Chord/Adapters/UdpInfoGridAdapter.java

package com.samsung.android.sdk.chord.example.adapter;

import java.util.ArrayList;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.samsung.android.sdk.chord.example.R;

public class UdpInfoGridAdapter extends ArrayAdapter<Item> {

    LayoutInflater mInflater;

    int layoutResourceId;

    ArrayList<Item> data = new ArrayList<Item>();

    public UdpInfoGridAdapter(Context context, LayoutInflater inflater, ArrayList<Item> data) {
        super(context, 0, data);

        this.mInflater = inflater;
        this.data = data;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View row = convertView;
        RecordHolder holder = null;

        if (row == null) {
            row = mInflater.inflate(R.layout.row_grid, null);
            holder = new RecordHolder();
            holder.txtTitle = (TextView) row.findViewById(R.id.item_text);
            holder.imageItem = (ImageView) row.findViewById(R.id.item_image);
            row.setTag(holder);
        } else {
            holder = (RecordHolder) row.getTag();
        }

        Item item = data.get(position);
        holder.txtTitle.setText(item.getText());
        holder.imageItem.setImageBitmap(item.getImage());

        return row;

    }

    static class RecordHolder {
        TextView txtTitle;

        ImageView imageItem;
    }

}

File: java2csharp/csharp/Samsung/Chord/UdpInfoGridAdapter.cs
using System.Collections.Generic;

namespace com.samsung.android.sdk.chord.example.adapter
{

	using Context = android.content.Context;
	using LayoutInflater = android.view.LayoutInflater;
	using View = android.view.View;
	using ViewGroup = android.view.ViewGroup;
	using ArrayAdapter = android.widget.ArrayAdapter;
	using ImageView = android.widget.ImageView;
	using TextView = android.widget.TextView;

	public class UdpInfoGridAdapter : ArrayAdapter<Item>
	{

		internal LayoutInflater mInflater;

		internal int layoutResourceId;

		internal List<Item> data = new List<Item>();

		public UdpInfoGridAdapter(Context context, LayoutInflater inflater, List<Item> data) : base(context, 0, data)
		{

			this.mInflater = inflater;
			this.data = data;
		}

		public override View getView(int position, View convertView, ViewGroup parent)
		{
			View row = convertView;
			RecordHolder holder = null;

			if (row == null)
			{
				row = mInflater.inflate(R.layout.row_grid, null);
				holder = new RecordHolder();
				holder.txtTitle = (TextView) row.findViewById(R.id.item_text);
				holder.imageItem = (ImageView) row.findViewById(R.id.item_image);
				row.Tag = holder;
			}
			else
			{
				holder = (RecordHolder) row.Tag;
			}

			Item item = data[position];
			holder.txtTitle.Text = item.Text;
			holder.imageItem.ImageBitmap = item.Image;

			return row;

		}

		internal class RecordHolder
		{
			internal TextView txtTitle;

			internal ImageView imageItem;
		}

	}
}